packages/plugins/plugin-table/src/next/hooks/useTableModel.ts

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type RefObject, useEffect, useMemo, useState } from 'react';
+
+import { type DxGridElement, type DxGridAxisMeta } from '@dxos/react-ui-grid';
+
+import { type TableType } from '../../types';
+import { TableModel } from '../table-model';
+
+// TODO(Zan): Take ordering here (or order based on some stored property).
+// When the order changes, we should notify the consumer.
+export const useTableModel = (table: TableType, data: any[], gridRef: RefObject<DxGridElement>) => {
```

Comment:
The model can be instantiated and used without react, which is what the unit tests in `table-model.test.ts`

Agree that it might be better for the query to move within the model itself, but I can't find an existing pattern for that anywhere. I've never managed a query without `useQuery` so would appreciate some guidance there.